### ISSUE TITLE
mscgen links placed in the wrong place when using SVG output

### ISSUE THREAD

**Issue by RedDwarf69** [doxygen/doxygen#6880]
This

```
/*! \mainpage
 * \msc
 *   arcgradient=16;
 *
 *   a [label="ElementA"],b [label="ElementB"], c [label="ElementC"];
 *   ...;
 *   --- [label="Lala"];
 *   |||;
 *   a->b [label="Message1", URL="@ref refa"];
 *   b=>c [label="Message2"];
 *   b<<c [label="Message3"];
 *   a<-b [label="Message4", URL="@ref refa"];
 *   ...;
 *   a<-b [label="Message5", URL="@ref refa"];
 *   a->b [label="Message6", URL="@ref refa"];
 *   a<-b [label="Message7", URL="@ref refa"];
 *   a->b [label="Message8", URL="@ref refa"];
 *   a->b [label="Message9", URL="@ref refa"];
 *   ...;
 *   a->b [label="Message10", URL="@ref refa"];
 * \endmsc
 *
 * # ReferenceA @anchor refa #
 */
```

Works fine with a default doxygen 1.8.15 generated Doxyfile (DOT_IMAGE_FORMAT=png). But when doing `( cat Doxyfile; echo DOT_IMAGE_FORMAT=svg ) | doxygen -` the hyperlinks on the svg image are not in the correct place.

Doxygen 1.8.15
graphviz 2.40.1
mscgen 0.20

**Comment by albert-github** [comment]
I had a look into the problem and with 'svg' the 'active' areas are not on the place as expected.
I extended the problem a little bit to get a better overview (unique references for each message and I also added references on the places where no references were present yet).
When looking at the map file generated this file is identical for 'png' and 'svg' type images
Based on this I manipulated the resulting 'svg' slightly so it shows the areas that should be the active areas: color red and the real active areas: color green. 
![result](https://user-images.githubusercontent.com/5223533/54143152-878ca480-4429-11e9-85c3-1a42abe6e420.jpg)

when looking at the png image (no coloring possible for me):
![result_png](https://user-images.githubusercontent.com/5223533/54143832-eef72400-442a-11e9-8466-52d88221b4c0.jpg)

we see that the green areas correspond with the text areas in the png image.
When looking in the mscgen 0.20 code we see that for the determination of the map there is no possibility to distinguish between the image types (the output type is in both case as follows `outType  = ADRAW_FMT_PNG`).

Possible solution:
- find the relation between the PNG and SVG coordinates
- parse the 'svg' file and the msc input and determine from the 'svg' file the real cordinates (the red parts in the first image).
- check if it is possible to
  - create an own mscgen executable with the output type in case ismap has the possibility of `outType  = ADRAW_FMT_PNG` or `outType  = ADRAW_FMT_SVG` depending on the the used file type
  - load the mapping process as a function directly into doxygen for svg
